assets/character/scripts/character_action_navigate_to.gd

```diff
@@ -1,14 +1,18 @@
 class_name CharacterActionNavigateTo extends CharacterAction
 
-var target_pos: Vector3
+var target_pos: Vector3:
+	set(pos):
+		target_pos = pos
+		if active:
+			character.navigation_agent.set_target_position(target_pos)
 
 func _init(owning_character: Character, pos: Vector3):
 	super._init(owning_character)
+	character.navigation_agent.navigation_finished.connect(complete)
 	target_pos = pos
 
 func start():
 	super.start()
-	character.navigation_agent.navigation_finished.connect(complete)
 	character.navigation_agent.set_target_position(target_pos)
```

Comment:
Init should not cause the action to start running, but putting the active check in the setter is incredibly confusing. Please stop going wild with the setters - it makes the code intractable as from a glance I have no idea whether something is going to cause whole bunch of side effects or I'm just setting a field.

Reply:
Ok, lets pair on how to do this cleaner